packages/hardhat-core/src/internal/hardhat-network/provider/fork/ForkTransaction.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  BufferLike,
+  PrefixedHexString,
+  Transaction,
+  TransactionOptions,
+  TxData,
+} from "ethereumjs-tx";
+import { BN, bufferToInt } from "ethereumjs-util";
+
+/**
+ * Custom Transaction class to avoid EIP155 errors when hardhat is forked
+ */
+export class ForkTransaction extends Transaction {
+  private readonly _chainId: number;
+
+  constructor(
+    chainId: number,
+    senderPubKey: Buffer,
+    data: Buffer | PrefixedHexString | BufferLike[] | TxData = {},
+    opts: TransactionOptions = {}
+  ) {
+    super(data, opts);
+
+    (this as any)._senderPubKey = senderPubKey;
+    this._chainId = chainId;
+  }
+
+  public verifySignature(): boolean {
+    return true;
+  }
+
+  public getChainId(): number {
+    return this._chainId;
+  }
+
+  public sign() {
+    throw new Error("Not implemented on ForkTransaction");
```

Comment:
This is temporary. Not sure which Hardhat Error to use here.

Reply:
I'd throw a ProviderError insteead of a hardhat error.